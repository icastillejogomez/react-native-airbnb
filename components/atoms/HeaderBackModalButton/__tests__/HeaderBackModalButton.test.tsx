import { render } from '@testing-library/react-native'

import { HeaderBackModalButton } from '..'

describe('HeaderBackModalButton', () => {
  test('renders without crashing', () => {
    const tree = render(<HeaderBackModalButton />)
    expect(tree).toMatchSnapshot()
  })

  test('run onPress handler when pressed', () => {
    const onPress = jest.fn()
    const tree = render(<HeaderBackModalButton onPress={onPress} />)
    tree.UNSAFE_getByType(HeaderBackModalButton).props.onPress()
    expect(onPress).toHaveBeenCalledTimes(1)
  })
})
