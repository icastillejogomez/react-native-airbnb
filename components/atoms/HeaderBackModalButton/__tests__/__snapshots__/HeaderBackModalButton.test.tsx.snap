// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBackModalButton renders without crashing 1`] = `null`;
